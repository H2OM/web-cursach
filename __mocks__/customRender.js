import ClientContext from '@/lib/context/ClientContext';
import { render } from '@testing-library/react';

export default customRender = (ui, {providerProps}) => {
    return render(
      <ClientContext.Provider value={{...providerProps}}>{ui}</ClientContext.Provider>,
    )
}