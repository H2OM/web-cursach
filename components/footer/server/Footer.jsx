import Miniform from '@/lib/basecomponents/miniform/miniform';
import '../footer.scss';
import Image from 'next/image';
import Link from 'next/link';
export default function Footer () {

    return (
        <footer className="Footer">
            <div className="container Footer__grid">
                <div className="Footer__block">
                    <Image
                        src="/img/logo_medium.png"
                        alt="logo"
                        className="Footer__logo"
                        width={0}
                        height={0}
                        sizes='100vw'
                        priority={true}
                        quality={100}
                    />
                    <div className="Footer__block__section">
                        <h3 className="subtitle">Контакты</h3>
                        <div className="Footer__block__section__desc">TopAttractions@gmail.com</div>
                        <div className="Footer__block__section__desc">+7 900 111-11-11</div>
                        <div className="Footer__block__section__icons">
                            <Image
                                src="/svg/icon_telegram.svg"
                                alt="tg"
                                className="Footer__block__section__icons__icon"
                                width={0}
                                height={0}
                                sizes='100vw'
                                priority={true}
                                quality={100}
                            />
                            <Image
                                src="/svg/icon_vk.svg"
                                alt="vk"
                                className="Footer__block__section__icons__icon"
                                width={0}
                                height={0}
                                sizes='100vw'
                                priority={true}
                                quality={100}
                            />
                        </div>
                        
                    </div>
                    <div className="Footer__block__section">
                        <h3 className="subtitle">Подпишитесь на новости</h3>
                        <Miniform/>
                    </div>
                </div>
                <div className="Footer__block Footer__block_second">
                    <Link className="Footer__block__link" href={"/"}>Политика конфиденциальности</Link>
                    <div className="Footer__block__section__desc" href={"/"}>© TopAttractions, 2024</div>
                </div>
            </div>
        </footer>
    )
}