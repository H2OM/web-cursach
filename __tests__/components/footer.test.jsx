import 'intersection-observer';
import customRender from '@/__mocks__/customRender';
import Footer from '@/components/footer/server/Footer';
import { screen } from '@testing-library/react';

it('footer snapshot', async () => {
    const {container} = customRender(<Footer/>, {});
    const images = screen.getAllByRole('img');
    const titles = screen.getAllByRole('heading');
    const info = screen.getAllByRole('info');
    expect(images.length).toBe(3);
    expect(titles.length).toBe(2);
    expect(info.length).toBe(3);
    expect(await screen.findByRole('link')).toBeInTheDocument();
    expect(await screen.getByText('Подписаться')).toBeInTheDocument();
    expect(await screen.getByPlaceholderText('Ваша почта')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
});