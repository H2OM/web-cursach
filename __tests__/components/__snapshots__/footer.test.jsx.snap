// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contacts snapshot 1`] = `
<div>
  <footer
    class="Footer"
  >
    <div
      class="container Footer__grid"
    >
      <div
        class="Footer__block"
      >
        <img
          alt="logo"
          class="Footer__logo"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="0"
          sizes="100vw"
          src="/_next/image?url=%2Fimg%2Flogo_medium.png&w=3840&q=100"
          srcset="/_next/image?url=%2Fimg%2Flogo_medium.png&w=640&q=100 640w, /_next/image?url=%2Fimg%2Flogo_medium.png&w=750&q=100 750w, /_next/image?url=%2Fimg%2Flogo_medium.png&w=828&q=100 828w, /_next/image?url=%2Fimg%2Flogo_medium.png&w=1080&q=100 1080w, /_next/image?url=%2Fimg%2Flogo_medium.png&w=1200&q=100 1200w, /_next/image?url=%2Fimg%2Flogo_medium.png&w=1920&q=100 1920w, /_next/image?url=%2Fimg%2Flogo_medium.png&w=2048&q=100 2048w, /_next/image?url=%2Fimg%2Flogo_medium.png&w=3840&q=100 3840w"
          style="color: transparent;"
          width="0"
        />
        <div
          class="Footer__block__section"
        >
          <h3
            class="subtitle"
          >
            Контакты
          </h3>
          <div
            class="Footer__block__section__desc"
            role="info"
          >
            TopAttractions@gmail.com
          </div>
          <div
            class="Footer__block__section__desc"
            role="info"
          >
            +7 900 111-11-11
          </div>
          <div
            class="Footer__block__section__icons"
          >
            <img
              alt="tg"
              class="Footer__block__section__icons__icon"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="0"
              src="/svg/icon_telegram.svg"
              style="color: transparent;"
              width="0"
            />
            <img
              alt="vk"
              class="Footer__block__section__icons__icon"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="0"
              src="/svg/icon_vk.svg"
              style="color: transparent;"
              width="0"
            />
          </div>
        </div>
        <div
          class="Footer__block__section"
        >
          <h3
            class="subtitle"
          >
            Подпишитесь на новости
          </h3>
          <form
            class="Miniform "
          >
            <div
              class="Miniform__input__wrap "
            >
              <input
                class="Miniform__input "
                name="mail"
                placeholder="Ваша почта"
                required=""
                type="text"
              />
              <label
                class="Miniform__input__error "
                for="email"
              >
                Неверный формат почты
              </label>
            </div>
            <button
              class="Miniform__button "
              type="submit"
            >
              Подписаться
            </button>
          </form>
        </div>
      </div>
      <div
        class="Footer__block Footer__block_second"
      >
        <a
          class="Footer__block__link"
          href="/"
        >
          Политика конфиденциальности
        </a>
        <div
          class="Footer__block__section__desc"
          href="/"
          role="info"
        >
          © TopAttractions, 2024
        </div>
      </div>
    </div>
  </footer>
</div>
`;
